### PR TITLE
Added documents to the file tabs | psp-4766

### PR DESCRIPTION
**Pull request by FuriousLlama**
image

Note: Also fixes PSP-4854

### CODE REVIEW

**Comment by devinleighsmith** [comment]
Noticing that the only way to get document updates to the documents tab is a hard refresh or by navigating away/back to the file. This may be a good use for the staleFile/setStaleFile property of the SideBarContext (That is how the activity updates are displayed).

**Comment by devinleighsmith** [comment]
Requirement: **Default is the most recent document uploaded** doesn't seem to be working.

**Comment by devinleighsmith** [comment]
Also the filter doesn't match the mockup: 
image

**Comment by devinleighsmith** [comment]
The font size also doesn't match up for the column text (16 vs 14), I wonder if that is a defect from the original document list view story or if Bobbi changed that for this story? might be worth checking with her.

File: source/backend/api/Areas/Documents/DocumentRelationshipController.cs
Comment (devinleighsmith) [inline source/backend/api/Areas/Documents/DocumentRelationshipController.cs:123]: yeah, I think the document controller makes more sense then the relationship controller.

File: source/backend/api/Services/DocumentActivityService.cs
Comment (devinleighsmith) [inline source/backend/api/Services/DocumentActivityService.cs:54]: FileT

File: source/backend/api/Services/DocumentService.cs
Comment (devinleighsmith) [inline source/backend/api/Services/DocumentService.cs:158]: github got a bit confused formatting these changes, but it seems like the main change is that the if statement at line 127 in the previous version was removed in favour of this if statement below. Can you help me understand the reasoning for that change? it seems like that would cause documentStorageRepository.GetDocumentMetadataAsync to get executed where it wouldn't have been previously.

File: source/backend/dal/Repositories/DocumentActivityRepository.cs
Comment (devinleighsmith) [inline source/backend/dal/Repositories/DocumentActivityRepository.cs:74]: duplicated include maybe?

File: source/frontend/src/features/documents/hooks/useDocumentProvider.ts
Comment (devinleighsmith) [inline source/frontend/src/features/documents/hooks/useDocumentProvider.ts:118]: Update document error or Update document metadata error?

File: source/frontend/src/features/documents/list/DocumentResults/DocumentResultsColumns.tsx
Comment (devinleighsmith) [inline source/frontend/src/features/documents/list/DocumentResults/DocumentResultsColumns.tsx:93]: I'd think this should be the last updated not the creator, but good to check requirements here.

**Review by devinleighsmith — APPROVED** [review]
No major issues, I did leave some comments that may need to be addressed before merging.

File: source/backend/dal/Repositories/DocumentActivityRepository.cs
Comment (FuriousLlama) [reply source/backend/dal/Repositories/DocumentActivityRepository.cs:74]: good catch!

File: source/frontend/src/features/documents/hooks/useDocumentProvider.ts
Comment (FuriousLlama) [reply source/frontend/src/features/documents/hooks/useDocumentProvider.ts:118]: Document. Although the call has metadata on its name, it is possible that the status is the only part being updated

File: source/frontend/src/features/documents/list/DocumentResults/DocumentResultsColumns.tsx
Comment (FuriousLlama) [reply source/frontend/src/features/documents/list/DocumentResults/DocumentResultsColumns.tsx:93]: I think is the creator given that the uploaded column is showing the date of creation

File: source/backend/api/Services/DocumentService.cs
Comment (FuriousLlama) [reply source/backend/api/Services/DocumentService.cs:158]: This was the cause of the bug. 
Thge line " foreach (var updateMetadata in updateRequest.DocumentMetadata)" was expecting that sometimes the updateRequest would not be empty or null before checking the metadata differences. This is a wrong assumption. The front end always sends a documentMetadata field array even if there are no changes.

Now it will perform the proper diff on the set and get the correct added/modified/deleted count. If the count of any of those  are not zero, then update them on Mayan and use that flag to mark the update to continue.

The GetDocumentMetadataAsync call is necessary regardless given our pattern is not to trust the front end to mark changes

File: source/backend/api/Areas/Documents/DocumentRelationshipController.cs
Comment (FuriousLlama) [reply source/backend/api/Areas/Documents/DocumentRelationshipController.cs:123]: The clear indication of this was that the relationshiptype was not being used at all in the call

**Comment by FuriousLlama** [comment]
> Noticing that the only way to get document updates to the documents tab is a hard refresh or by navigating away/back to the file. This may be a good use for the staleFile/setStaleFile property of the SideBarContext (That is how the activity updates are displayed).

That worked like a charm. Thanks!

**Comment by FuriousLlama** [comment]
@devinleighsmith Regarding the size, default drop size and Filter by. I did not change those given it could possibly affect all instances of the component of the document. I will change it with Bobbie to make sure that is understood. Alternatively, doing changes that are only needed on a single screen would increase the complexity of the component without (maybe) a strong reason for it.

**Comment by devinleighsmith** [comment]
> @devinleighsmith Regarding the size, default drop size and Filter by. I did not change those given it could possibly affect all instances of the component of the document. I will change it with Bobbie to make sure that is understood. Alternatively, doing changes that are only needed on a single screen would increase the complexity of the component without (maybe) a strong reason for it.

The above makes sense to me, but we want to make sure that we get signoff from Bobbi or we at least declare these known issues to Sue, otherwise these decisions will come back as bugs.

**Pull request merged**
